skll/utilities/generate_predictions.py

```diff
@@ -169,8 +205,34 @@ def main(argv=None):
                                                     label_col=args.label_col,
                                                     id_col=args.id_col)
             feature_set = reader.read()
-            for pred in predictor.predict(feature_set):
-                print(pred)
+            preds = predictor.predict(feature_set)
+            header = predictor.output_file_header
+
+            if args.output_file is not None:
+                with open(args.output_file, 'a') as outputfh:
+                    if i == 0:  # Only write header once per set of input files
+                        print("\t".join(header), file=outputfh)
+                    if args.all_probabilities:
+                        for i, probabilities in enumerate(preds):
+                            id_ = feature_set.ids[i]
+                            probs_str = "\t".join([str(p) for p in probabilities])
+                            print("{}\t{}".format(id_, probs_str), file=outputfh)
+                    else:
+                        for i, pred in enumerate(preds):
```

Comment:
Can we call this `j` or something else, since we're using `i` in the outer loop? Same comment below.